### ISSUE TITLE
How do you use Mustache with RequireJS?

### ISSUE THREAD

**Issue by remino** [janl/mustache.js#528]
I'd like a simple explanation on how to use Mustache with RequireJS.

I've been trying to load it in my code. The `mustache.js` is getting loaded, but `Mustache` or `mustache` are never defined thus not usable by the code that requires Mustache.

I'm stuck having to use the jQuery plugin version, but I'd rather use Mustache without having to rely on jQuery.

Also, an explanation on how to do so in the `README` or in the wiki would be very helpful.

**Comment by phillipj** [comment]
That should be very simple as mustache.js supports AMD modules.

Lets say you've got mustache.js placed in `./libs/mustache.js`, then your `main.js` could look something like this:
```js
require.config({
  paths: {
    'mustache': 'libs/mustache'
  }
});

requirejs(['mustache'], function(mustache) {
  console.log('mustache exists:', mustache !== undefined);
});
```

Did that answer your question?

**Comment by rndme** [comment]
fwiw, just loading the main script (not as a module, just regular) will expose the Mustache global, no need for jq...

**Comment by remino** [comment]
I figured it can't be that difficult, and looking at the wrapper for Mustache, it will check if RequireJS exists or not and load itself accordingly.

What @phillipj suggested should work, but I used [requirejs-rails](https://github.com/jwhitley/requirejs-rails) to load RequireJS. The config is passed to JavaScript from the config in a YAML file. It should work the same, but it didn't. Of course, @rndme is right, jQuery isn't necessary. But somehow that was the only way to make it work in my case.

Since then, my project requirements changed and I no longer need Mustache or RequireJS. Perhaps it was just something I did wrong. I'll close the comment for now, as I feel it's no longer relevant.
